### PR TITLE
GraphQL - FancyLoader modify_query enhancement

### PR DESCRIPTION
**Pull request by toyhammered**
# What

Allow modify_query to be added as an option when creating a FancyLoader.

# Why

If we use the RankedModel `*_rank` methods provided by RankedModel the sql query will be absolute trash. We needed a way to optimize this, and be more human readable (`0..`) instead of these obscurely large numbers. With the addition of `modify_query` we are now able to modify the query before execution.

# Checklist

- [x] All files pass Rubocop
- [x] Any complex logic is commented
- [x] Any new systems have thorough documentation
- [x] Any user-facing changes are behind a feature flag (or: explain why they can't be)

- [x] All the tests pass
- [x] Tests have been added to cover the new code

### CODE REVIEW

File: app/graphql/loaders/fancy_loader/rank_query_generator.rb
Comment (NuckChorris) [inline app/graphql/loaders/fancy_loader/rank_query_generator.rb]: I don't think `key` is the right name for this, how about

```suggestion
  # @param column [Symbol] The column to rank by
  # @param partition_key [Symbol] The find_by key for the table
  # @param table [Arel::Table]
  def initialize(column, partition_key, table)
    @column = column
```

File: app/graphql/loaders/fancy_loader/rank_query_generator.rb
Comment (NuckChorris) [inline app/graphql/loaders/fancy_loader/rank_query_generator.rb]:
```suggestion
    "#{@column}_rank"
```

File: app/graphql/loaders/fancy_loader/rank_query_generator.rb
Comment (NuckChorris) [inline app/graphql/loaders/fancy_loader/rank_query_generator.rb]:
```suggestion
    @table[@column].asc
```

File: app/graphql/loaders/installments_loader.rb
Comment (NuckChorris) [inline app/graphql/loaders/installments_loader.rb]: This might look nicer if you switch to keyword arguments 🤔 Not 100% sure though, feel free not to

```ruby
release_rank = Loaders::FancyLoader::RankQueryGenerator.new(
  column: :release_order,
  table: table,
  partition_by: @find_by
).arel
```

File: app/graphql/types/installment.rb
Comment (NuckChorris) [inline app/graphql/types/installment.rb:14]: I think a comment on these might be nice like

```ruby
    # Try preloaded rank first, fall back to ranked-model's slower COUNT(*) query
    object.attributes['release_order_rank'] || object.release_order_rank
```

**Review by NuckChorris — COMMENTED** [review]
Minor nits, overall this looks great!

**Review by NuckChorris — APPROVED** [review]
LGTM

**Comment by toyhammered** [comment]
Will merge once the base branch has been approved

**Pull request merged**
